SPECS/netdata.spec

```diff
@@ -1,391 +1,464 @@
+install -m 600 %{SOURCE1004} %{buildroot}%{_sysconfdir}/sysconfig/ip6tables_netdata
+
+# Scripts must not be in /etc, /usr/libexec is a better place
+mv %{buildroot}%{_sysconfdir}/%{name}/edit-config %{buildroot}%{_libexecdir}/%{name}/edit-config
+# Fix EOL
+sed -i -e 's/\r//' %{buildroot}%{_datadir}/%{name}/web/lib/tableExport-1.6.0.min.js
+# Delete system dir with init scripts or unit files
+rm -rf %{buildroot}%{_libdir}/%{name}/system
+# Delete useless hidden dir
+rm -rf %{buildroot}%{_datadir}/%{name}/web/.well-known
+# Delete useless file (ubuntu)
+rm -f %{buildroot}%{_sysconfdir}/%{name}/conf.d/ebpf.d/ebpf_kernel_reject_list.txt
+
+for dir in charts.d health.d python.d statsd.d ; do
+  mkdir -p %{buildroot}%{_sysconfdir}/%{name}/${dir}
+done
+
+mkdir -p %{buildroot}%{_sysconfdir}/profile.d
+install -p -m 0644 %{SOURCE4} %{buildroot}%{_sysconfdir}/profile.d/netdata.sh
+sed -i -e '/NETDATA_STOCK_CONFIG_DIR/s;@STOCK_CONFIG_DIR@;%{netdata_conf_stock};' %{buildroot}%{_sysconfdir}/profile.d/netdata.sh
+
+# Integrate go plugins
+mkdir -p %{buildroot}%{_sysconfdir}/%{name}/go.d
+install -p conf.d/go.d.conf %{buildroot}%{netdata_conf_stock}/conf.d/go.d.conf
+cp -rp conf.d/go.d %{buildroot}%{netdata_conf_stock}/conf.d/go.d
+install -p -m 0644 packaging/go.d.checksums %{buildroot}%{_datadir}/%{name}/go.d.checksums
+install -p -m 0750 %{SOURCE21} %{buildroot}%{_sbindir}/netdata-install-go-plugins.sh
+sed -i \
+    -e 's;@PLUGIN_GO_VERSION@;%{plugin_go_ver};' \
+    -e 's;@DATADIR@;%{_datadir};' \
+    -e 's;@LIBEXEC@;%{_libexecdir};' \
+    %{buildroot}%{_sbindir}/netdata-install-go-plugins.sh
+    
+rm -f %{buildroot}%{_sysconfdir}/%{name}/netdata-updater.conf
+
+%check
+make tests
+
+%pre data
+getent group netdata > /dev/null || groupadd -r netdata
+getent passwd netdata > /dev/null || useradd -r -g netdata -c "NetData User" -s /sbin/nologin -d /var/log/%{name} netdata
 
 %post
 if [ $1 -eq 1 ]; then
-    ln -s netdata.conf.headless /etc/netdata/netdata.conf
     # Disable telemetry by default on first install
     touch /etc/netdata/.opt-out-from-anonymous-statistics
+
+    # Make netdata.conf a symlink to the disabled webui config file on first install
+    ln -s netdata.conf.headless /etc/netdata/netdata.conf
+fi
+# If /etc/netdata/netdata.conf is a symlink, then it is the xcp-ng specific version already containing
+# correct values and there's no need to update it
+if [ ! -L /etc/netdata/netdata.conf ]; then
+    sed -i -e '/web files group/ s/root/netdata/' /etc/netdata/netdata.conf ||:
+    sed -i -e '/stock config directory/ s;/etc/netdata/conf.d;/usr/lib/netdata/conf.d;' /etc/netdata/netdata.conf ||:
+    sed -i -e '/stock health configuration directory/ s;/etc/netdata/conf.d/health.d;/usr/lib/netdata/conf.d/health.d;' /etc/netdata/netdata.conf ||:
 fi
-%{netdata_init_post}
+%systemd_post %{name}.service
+/usr/bin/systemctl enable %{name}.service
+/usr/bin/systemctl daemon-reload
+/usr/bin/systemctl restart %{name}.service
```

Comment:
%systemd_post is made obsolete what the next 3 lines.
```suggestion

# XCP-ng: always enable and start the service
/usr/bin/systemctl enable %{name}.service
/usr/bin/systemctl daemon-reload
/usr/bin/systemctl restart %{name}.service
```

Another comment, but that's not specific to this PR: the day when we start bundling netdata at installation time, or that users start doing it on their side, this will likely fail. We should probably at least add `||:` at the end of each line so that it doesn't break the installation transaction.

Reply:
Done